### PR TITLE
Refactor/kol get mentions

### PR DESCRIPTION
**Pull request by Xm0onh**
This PR is about moving thread of conversation from `responseGenerationNode` to `mentionNode`. Now, the thread is also being used in `engagementNode`, `toneAnalysisNode` and "responseGenerationNode`.
The algorithm of fetching mentioned modified and improved by `n` factor.

- The changes in this PR will effect memory viewer. I'll open a seperate PR for that.
- It would be great to craft the prompts further to take the thread into account properly.
- One of the nice thing to have is to summarize a conversation thread and extract the important parts and use them instead of passin the whole thread as a prompt. This change will make agent more efficient when we are dealing with large threads.

### CODE REVIEW

File: auto-kol/agent/src/services/twitter/api.ts [hunk 1]
```diff
@@ -101,63 +101,130 @@ export class ExtendedScraper extends Scraper {
         return replies;
     }
 
-    async getThread(tweetId: string): Promise<Tweet[]> {
-        const username = config.TWITTER_USERNAME!;
-        const isLoggedIn = await this.isLoggedIn();
+    async getThread(tweetId: string, maxDepth: number = 100): Promise<Tweet[]> {
```
Comment (jfrank-summit) [inline auto-kol/agent/src/services/twitter/api.ts:104]: `maxDepth` is not used

File: auto-kol/agent/src/services/twitter/api.ts [hunk 2]
```diff
@@ -101,63 +101,130 @@ export class ExtendedScraper extends Scraper {
...
+                    isLoggedIn = await this.isLoggedIn();
+                    if (isLoggedIn) {
+                        logger.info('Successfully re-authenticated');
+                        break;
+                    }
+                    logger.error('Re-authentication failed');
+                    retryCount++;
+                    if (retryCount < maxRetries) {
+                        // Wait for 2 seconds before retrying
+                        await new Promise(resolve => setTimeout(resolve, 2000));
```
Comment (jfrank-summit) [inline auto-kol/agent/src/services/twitter/api.ts:124]: We should use an exponential backoff.

File: auto-kol/agent/src/services/agents/nodes/mentionNode.ts [hunk 3]
```diff
@@ -3,8 +3,10 @@ import { parseMessageContent, WorkflowConfig } from "../workflow.js";
 import { logger } from "../workflow.js";
 import { State } from "../workflow.js";
 import { tweetSearchSchema } from "../../../schemas/workflow.js";
+import { ExtendedScraper } from "../../twitter/api.js";
+import { Tweet } from "../../../types/twitter.js";
 
-export const createMentionNode = (config: WorkflowConfig) => {
+export const createMentionNode = (config: WorkflowConfig, scraper: ExtendedScraper) => {
```
Comment (jfrank-summit) [inline auto-kol/agent/src/services/agents/nodes/mentionNode.ts:9]: `scraper` already exists on `config` as `config.client`.

File: auto-kol/agent/src/services/twitter/api.ts [hunk 4]
```diff
@@ -101,63 +101,130 @@ export class ExtendedScraper extends Scraper {
...
+                    }
+                }
+            }
+            
+            if (!isLoggedIn) {
+                logger.error(`Failed to re-authenticate after ${maxRetries} attempts`);
+                return [];
```
[the remark this answers is not in the document]
Comment (Xm0onh) [reply auto-kol/agent/src/services/twitter/api.ts:138]: Yup, at least it didn't happen in other places.

**Comment by Xm0onh** [comment]
@jfrank-summit I discovered a bug in my getThread implementation. For now, let’s keep my version as a placeholder. I’ve also added the previous function as the origin function for fetching a conv thread.

Let’s prioritize merging this PR since retrieving the thread has been moved to the start of the workflow. I have a more urgent task to address and will revisit this optimization later.

**Comment by Xm0onh** [comment]
@jfrank-summit conflicts are resolved.

**Review by jfrank-summit — APPROVED** [review]
LGTM

**Pull request merged**